addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1533,25 +1533,30 @@ export class Wysiwyg extends Component {
         const targetEl = this.odooEditor.document.getSelection();
         const closest = closestBlock(targetEl.anchorNode);
         const restoreSelection = preserveCursor(this.odooEditor.document);
-        const blockWidth = parseInt(getComputedStyle(closest).width.slice(0, -2)); // remove 'px'
-        // Calculate one/fouth portion of blockWidth, and then on the basis of our cursor position
-        // check in which quadrant it falls, give the popover its position accordingly
-        const portionWidth = (blockWidth / 4);
-        const cursorAt = getRangePosition(targetEl, this.odooEditor.document).left;
-        let position = 'end';
-        if (cursorAt < portionWidth * 2) {
-            position = 'start';
-        } else if (cursorAt <= portionWidth * 3) {
-            position = 'middle';
-        }
 
         this.popover.add(closest, EmojiPicker, {
                 onSelect: (str) => {
                     restoreSelection();
                     this.odooEditor.execCommand('insert', str);
                 },
+            }, {
+                onPositioned: (popover, position) => {
+                    restoreSelection();
+                    const rangePosition = getRangePosition(popover, this.options.document, this.options.getContextFromParentRect());
```

Comment:
Hello! The position of the emoji picker seems broken in mass_mailing.

I suspect it's because of a problem with `options.document` in mass_mailing's Wysiwyg, that gets (wrongly) assigned to the top document instead of the iframe's document after the Wysiwyg component is re-rendered.
Considering that https://github.com/odoo/odoo/pull/138259, which aims to fix it, was recently merged into saas-16.4, you might wanna give it a little test after rebasing your branch.

Alternatively, you could use `this.odooEditor.document` instead of `this.options.document` here.